post-processor/vsphere/post-processor.go

```diff
@@ -136,6 +140,43 @@ func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (pac
 		return nil, false, fmt.Errorf("Failed: %s\n", err)
 	}
 
+	if p.config.ConvertToTemplate {
+		sdkURL, err := url.Parse(fmt.Sprintf("https://%s:%s@%s/sdk",
+			url.QueryEscape(p.config.Username),
+			url.QueryEscape(p.config.Password),
+			p.config.Host))
+
+		if err != nil {
+			return nil, false, fmt.Errorf("Failed: %s\n", err)
+		}
+
+		client, err := govmomi.NewClient(context.TODO(), sdkURL, true)
+
+		if err != nil {
+			return nil, false, fmt.Errorf("Failed: %s\n", err)
+		}
+
+		finder := find.NewFinder(client.Client, false)
+		datacenter, err := finder.DefaultDatacenter(context.TODO())
+		finder.SetDatacenter(datacenter)
+		if err != nil {
+			return nil, false, fmt.Errorf("Failed: %s\n", err)
+		}
+
+		vm, err := finder.VirtualMachine(context.TODO(), p.config.VMName)
+
+		if err != nil {
+			return nil, false, fmt.Errorf("Failed: %s\n", err)
+		}
+
+		ui.Message(fmt.Sprintf("Marking as template %s", p.config.VMName))
+		err = vm.MarkAsTemplate(context.TODO())
```

Comment:
`context.TODO()` might actually be fine but I would default to `context.Background()`.